### PR TITLE
DM-27811: Remove deprecated filterLabel component.

### PR DESCRIPTION
**Pull request by erykoff**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: doc/changes/DM-27811.removal.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove deprecated filterLabel exposure component access.
```
Comment (kfindeisen) [inline doc/changes/DM-27811.removal.md:1]: I would argue for merging these two commits, as they're not self-contained -- if 2e0ca30 were removed, this log would be incorrect; if 4877849 were removed, the cleanup would be incomplete.

**Review by kfindeisen — CHANGES_REQUESTED** [review]
Unfortunately, you missed `exposureAssembler.py`, which needs to be kept in sync with the non-disassembled code. The way the tests are set up means that removing `filterLabel` from `test_butlerFits` prevents the mismatch from being tested. :disappointed:

**Comment by erykoff** [comment]
Thanks for catching that!  I was relying too much on todos and not enough on grep.  Anyway, everything is removed now.

**Review by kfindeisen — APPROVED** [review]
Looks good!

**Pull request merged**